### PR TITLE
:sparkles:Add cluster provider for KCP

### PR DESCRIPTION
**Pull request by andreyod**
## Summary
Cluster provider for KCP runtime

## Related issue(s)

Fixes #

### CODE REVIEW

File: clusterprovider/kcp/kcp.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+	"k8s.io/klog/v2"
+
+	kcpv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/core/v1alpha1"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	tenancyclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned/typed/tenancy/v1alpha1"
+
+	clusterprovider "github.com/kubestellar/kubestellar/pkg/clustermanager/providerclient"
+)
+
+// KcpClusterProvider is a cluster provider that works with a local Kind instance.
```
Comment (ezrasilvera) [inline clusterprovider/kcp/kcp.go:41]: KIND in comment

File: clusterprovider/kcp/kcp.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+	workspaces map[string]string
+	watch      clusterprovider.Watcher
+}
+
+// New returns a new KcpClusterProvider
+func New() (*KcpClusterProvider, error) {
+	ctx := context.Background()
+	logger := klog.FromContext(ctx)
+
+	kcpConfigPath := os.Getenv("HOME") + "/.kcp/config"
```
Comment (ezrasilvera) [inline clusterprovider/kcp/kcp.go:56]: config to the cluster should be passed not loaded

File: clusterprovider/kcp/kcp.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+		configPath: kcpConfigPath,
+		clientset:  clientset.Workspaces(),
+		workspaces: make(map[string]string),
+	}
+	return c, nil
+}
+
+func (k *KcpClusterProvider) Create(name string, opts clusterprovider.Options) error {
+	k.logger.V(2).Info("Creating KCP workspace", "name", name)
+	ws := &tenancyv1alpha1.Workspace{
```
Comment (ezrasilvera) [inline clusterprovider/kcp/kcp.go:84]: Where do we create it? under root?

File: clusterprovider/kcp/kcp.go [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+func (k *KcpClusterProvider) Delete(name string, opts clusterprovider.Options) error {
+	k.logger.V(2).Info("Deleting KCP workspace", "name", name)
+	return k.clientset.Delete(k.ctx, name, metav1.DeleteOptions{})
+}
+
+// ListClustersNames is N/A for KCP
+// TODO remove it from ProviderClient interface
+func (k KcpClusterProvider) ListClustersNames() ([]string, error) {
+	return []string{}, nil
```
Comment (ezrasilvera) [inline clusterprovider/kcp/kcp.go:106]: Please print "Not implemented"  until we remove it

**Comment by andreyod** [comment]
Closing in favor of https://github.com/kubestellar/kubestellar/pull/849

**Pull request closed without merging**